code/messaging/src/main/java/com/adobe/marketing/mobile/PropositionItem.java

```diff
@@ -124,6 +133,49 @@ Inbound decodeContent() {
         return inboundContent;
     }
 
+    /**
+     * Creates a {@code PropositionItem} object from the provided {@code Map<String, Object>}.
+     *
+     * @param eventData {@link Map<String, Object>} event data
+     * @return {@link PropositionItem} object created from the provided {@code Map<String, Object>}.
+     */
+    static PropositionItem fromEventData(final Map<String, Object> eventData) {
+        final String uniqueId = DataReader.optString(eventData, PAYLOAD_ID, "");
+        final String schema = DataReader.optString(eventData, PAYLOAD_SCHEMA, "");
+        final Map<String, Object> dataMap = DataReader.optTypedMap(Object.class, eventData, PAYLOAD_DATA, null);
+        final Map<String, Object> contentMap = DataReader.optTypedMap(Object.class, dataMap, PAYLOAD_CONTENT, null);
```

Comment:
Content will not always be an Object map. We should create a JSONObject from content map if schema is json-content-item and then serialize it using toString().

Reply:
Update toEventData() implementation too based on this.